src/library/scala/collection/generic/IsMapLike.scala

```diff
@@ -0,0 +1,66 @@
+package scala.collection
+package generic
+
+import IsMapLike.Tupled
+
+/**
+  * Type class witnessing that a collection type `C`
+  * has keys of type `K`, values of type `V` and has a conversion to `MapOps[K, V, _, _]`.
+  *
+  * This type enables simple enrichment of `Map`s with extension methods.
+  *
+  * @tparam Repr Collection type (e.g. `Map[Int, String]`)
+  */
+trait IsMapLike[Repr] extends IsIterableLike[Repr] {
```

Comment:
It's not clear to me what functionality this adds. `IsIterableLike` and `IsSeqLike` can abstract over `String` and `Array` in addition to collections. There is no such case for maps.

Reply:
There is `MapView` that doesn’t extend `Map`.